### ISSUE TITLE
Automation visual editor does not populate required data 'zone'

### ISSUE THREAD

**Issue by dougclarknc** [home-assistant/frontend#20990]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Automation visual editor not including zone in trigger based on device enter/leaving zone despite visual editor indicating zone radio button has been selected
Editing in yaml allows user to add zone data required and save as expected

### Describe the behavior you expected

Expected zone data to be added to service call as indicated visually by editor

### Steps to reproduce the issue

Given: one zone has been defined AND a device_tracker is defined and capable of location services
Given: the automation visual editor is open
When: trigger is added, based on device, location enabled device is selected as target device, and trigger "device leaves a zone" is selected
Then: radio button list displayed showing only defined zone, by default only defined zone is displayed as selected
When: user clicks save
Expected: save button disappears, toast confirming save appears
Actual: toast error "Message malformed: required key not provided @ data['zone']"
When: user selects "edit in yaml"
Expected: 
```
  - platform: device
    device_id: ###
    domain: device_tracker
    entity_id: device_tracker.device
    type: leaves
    zone: zone.home
```
Actual:
  ```
- platform: device
    device_id: ###
    domain: device_tracker
    entity_id: ###
    type: leaves
```

### What version of Home Assistant Core has the issue?

2024.5.5

### In which browser are you experiencing the issue with?

Firefox developer 127.0b8 (64-bit), Chrome Version 125.0.6422.141 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

Win11 Home 22621.3672

**Comment by karwosts** [comment]
This will be fixed in 2024.6 by #20810

**Comment by dougclarknc** [comment]
@karwosts thank you! i didnt find 20810 or 118427 when searching
